components/prism-wgsl.js

```diff
@@ -0,0 +1,94 @@
+        greedy: true,
+    },
+    'attributes': {
+        pattern: /(@)(?:(?:[_\p{XID_Start}][\p{XID_Continue}]+)|(?:[\p{XID_Start}]))/u,
+        lookbehind: true,
+        alias: 'symbol',
+    },
+    'functions': {
+        pattern: /\b(fn)\s+(?:(?:[_\p{XID_Start}][\p{XID_Continue}]+)|(?:[\p{XID_Start}]))(?=(?:\()|(?:<))/u,
+        lookbehind: true,
+        alias: 'function',
+        inside: {},
+    },
+    'keyword': [
+        { pattern: /\b(?:bitcast|block|break|case|continue|continuing|default|discard|else|elseif|enable|fallthrough|for|function|if|loop|private|read|read_write|return|storage|switch|uniform|workgroup|write)\b/ },
+        { pattern: /\b(?:asm|const|do|enum|handle|mat|premerge|regardless|typedef|unless|using|vec|void|while)\b/ },
+        { pattern: /\b(?:let|var)\b/ },
+        { pattern: /\b(?:type)\b/ },
+        { pattern: /\b(?:enum)\b/ },
+        { pattern: /\bfn\b/ },
+    ],
+    'function-calls': {
+        pattern: /(?:(?:[_\p{XID_Start}][\p{XID_Continue}]+)|(?:[\p{XID_Start}]))(?=\()/u,
+        alias: 'function',
+        inside: {},
+    },
+    'types': [
+        { pattern: /\b(?:bool|i32|u32|f32)\b/, alias: 'builtin' },
+        { pattern: /\b(?:i64|u64|f64)\b/, alias: 'builtin' },
+        { pattern: /\b(?:vec[2-4]|mat[2-4]x[2-4])\b/, alias: 'builtin' },
+        { pattern: /\b(?:atomic)\b/, alias: 'builtin' },
+        { pattern: /\b(?:array)\b/, alias: 'builtin' },
+        { pattern: /\b(?:override|ptr|sampler|sampler_comparison|staticAssert|struct)\b/, alias: 'builtin' },
+        { pattern: /\b(?:texture_1d|texture_2d|texture_2d_array|texture_3d|texture_cube|texture_cube_array|texture_multisampled_2d)\b/, alias: 'builtin' },
+        { pattern: /\b(?:texture_storage_1d|texture_storage_2d|texture_storage_2d_array|texture_storage_3d|texture_depth_2d)\b/, alias: 'builtin' },
+        { pattern: /\b(?:texture_depth_2d_array|texture_depth_cube|texture_depth_cube_array|texture_depth_multisampled_2d)\b/, alias: 'builtin' },
+        { pattern: /\b(?:[A-Z][A-Za-z0-9]*)\b/, alias: 'class-name' },
+    ],
+    'built-in-values': [
+        { pattern: /\b(?:vertex_index|instance_index|position)\b/, alias: 'namespace' },
+        { pattern: /\b(?:front_facing|frag_depth|sample_index|sample_mask)\b/, alias: 'namespace' },
+        { pattern: /\b(?:local_invocation_id|local_invocation_index|global_invocation_id|workgroup_id|num_workgroups)\b/, alias: 'namespace' },
+    ],
+    'bool-literal': {
+        pattern: /\b(?:true|false)\b/,
+        alias: 'boolean',
+    },
+    'hex-int-literal': [
+        { pattern: /\b0[xX][0-9a-fA-F]+[iu]?(?!(?:\.|p|P))\b/, alias:'number' },
+    ],
+    'hex-float-literal': [
+        { pattern: /0[xX][0-9a-fA-F]*\.[0-9a-fA-F]+(?:[pP](?:\+|-)?[0-9]+[fh]?)?/, alias:'number' },
+        { pattern: /0[xX][0-9a-fA-F]+\.[0-9a-fA-F]*(?:[pP](?:\+|-)?[0-9]+[fh]?)?/, alias:'number' },
+        { pattern: /0[xX][0-9a-fA-F]+[pP](?:\+|-)?[0-9]+[fh]?/, alias:'number' },
```

Comment:
Prism isn't a parser or compiler. We are a syntax highlighter. This means that we can do some things parsers can't do. E.g. ignore invalid syntax.

Prism is allowed to produce any highlighting for invalid syntax. This allows us to simplify many regexes.

This regex can also be simplified.

```suggestion
        { pattern: /0[xX][0-9a-fA-F]*(?:\.[0-9a-fA-F]*)?(?:[pP][+-]?\d+[fh]?)?/, alias:'number' },
```